optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/Joiners.java

```diff
@@ -89,6 +109,18 @@
         return new SingleBiJoiner<>(leftMapping, JoinerType.GREATER_THAN_OR_EQUAL, rightMapping);
     }
 
+    /**
+     * Applies a filter to the joined tuple, with the semantics of {@link BiConstraintStream#filter(BiPredicate)}.
+     *
+     * @param filter filter to apply
+     * @param <A> type of the first fact in the tuple
+     * @param <B> type of the second fact in the tuple
+     * @return the joiner
```

Comment:
never null or sometimes null?

Reply:
I'd not even mention "the joiner" as the return type already signals that information.